src/activity_actor.cpp

```diff
@@ -2351,6 +2351,50 @@ void reload_activity_actor::finish( player_activity &act, Character &who )
     } else {
         add_msg( m_neutral, _( "You reload the %1$s with %2$s." ), reloadable_name, ammo_name );
     }
+    item_location loc = reload_targets[0];
+    tripoint point = loc.position();
+    // Reload may have caused the item to increase in size more than the pocket can contain.
+    // We want to avoid this because items will be deleted on a save/load.
+    if( loc.where() == item_location::type::container &&
+        loc.held_by( who ) ) {
+        item_pocket pocket = reloadable.get_containing_pocket( who );
```

Comment:
`reloadable` comes from `reload_targets`, which has `item_location` type. `item_location` allows to get `parent` directly. When you have parent, you can use `item::contained_where` to find parent's pocket where item is contained.

Reply:
Will look into that.

Reply:
Ok have removed the get_containing_pocket() addition and changed it to use the item_location. It makes more sense; I should know better.